### PR TITLE
ARROW-9353: [Python][CI] Disable known failures in dask integration tests

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Also removed the previously disabled tests (ARROW-5730 / from https://github.com/apache/arrow/pull/4718), because in the meantime those should be fixed.

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
@github-actions crossbow submit test-conda-python-3.7-dask-latest test-conda-python-3.8-dask-master

**Comment by jorisvandenbossche** [comment]
So both are now only failing for the ``test_illegal_column_name`` test (which I didn't skip yet as I don't understand why it is failing)

**Comment by jorisvandenbossche** [comment]
@github-actions crossbow submit test-conda-python-3.7-dask-latest test-conda-python-3.8-dask-master

File: ci/scripts/integration_dask.sh [hunk 1]
```diff
@@ -32,7 +32,11 @@ python -c "import dask.dataframe"
...
-pytest -v --pyargs dask.dataframe.tests.test_dataframe
+# skip failing pickle test, see https://github.com/dask/dask/issues/6374
+pytest -v --pyargs dask.dataframe.tests.test_dataframe -k "not test_dataframe_picklable"
 pytest -v --pyargs dask.dataframe.io.tests.test_orc
+# skip failing parquet tests, see https://github.com/dask/dask/issues/6243
+# test_illegal_column_name can be removed once next dask release is out
+# (https://github.com/dask/dask/pull/6378)
 pytest -v --pyargs dask.dataframe.io.tests.test_parquet \
-  -k "not test_timestamp_index and not test_roundtrip_from_pandas"
+  -k "not test_to_parquet_pyarrow_w_inconsistent_schema_by_partition_fails_by_default and not test_timeseries_nulls_in_schema and not test_illegal_column_name"
```
Comment (jorisvandenbossche) [inline ci/scripts/integration_dask.sh:42]: Not familiar enough with sh to know if there is a way to split this string on multiple lines (but also not the most important issue I think for this bash script)

Comment (pitrou) [reply ci/scripts/integration_dask.sh:42]: Not important IMO.

**Review by pitrou — APPROVED** [review]
+1

**Pull request closed without merging**